gmso/utils/conversions.py

```diff
@@ -137,3 +138,77 @@ def convert_ryckaert_to_opls(ryckaert_connection_type):
     )
 
     return opls_connection_type
+
+
+def check_convert_kelvin_to_energy_units(
+    energy_input_unyt,
+    energy_output_unyt_units_str,
+):
+    """Convert the Kelvin (K) energy unit to a standard energy unit.
+
+    Check to see if the unyt energy value is in Kelvin (K) and converts it to
+    another energy unit (Ex: kcal/mol, kJ/mol, etc.).  Otherwise, it passes thru the
+    existing unyt values.
+
+    Parameters
+    ----------
+    energy_input_unyt : unyt.unyt_quantity
+        The energy in units of 'energy / other_units' Example: 'energy/mol/angstroms**2' or 'K/angstroms**2'.
+        NOTE: The only valid temperature unit for thermal energy is Kelvin (K).
+    energy_output_unyt_units_str : str (unyt valid energy units only)
```

Comment:
Cal.  Good thought here.  Yes, kJ and kcal are accepted here, since Unyt does not really consider mol a unit.  I added the following as a comment to the doc string:

NOTE and WARNING: the energy units of kcal, kJ will be accepted due to the way the Unyt module does not accept mol as a recorded unit; however, this will result in  incorrect results from the intended function.

Reply:
I am unsure if we should do any more than this.  Thoughts?